### DISCUSSION TITLE
[2024-01-22] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by Frizi** [enso-org/enso#8824] [Weekly Demo Day]
@Frizi

- Finished code reordering on node connection https://github.com/enso-org/enso/issues/8494
- Added support for argument placeholder widgets https://github.com/enso-org/enso/issues/8788
- Fixed various other cases of argument placeholder lists being incorrectly generated for partially applied functions

<img width="1241" alt="image" src="https://github.com/enso-org/enso/assets/919491/5377f57f-18f0-4a50-a8ab-9331862ca547">

@farmaazon 

- Addressing various GUI bugs https://github.com/enso-org/enso/pull/8749
- Creating more E2E Test scenarios https://github.com/enso-org/enso/issues/8518

**Comment by kazcw** [comment]
- Implemented Y.Js storage of all AST data (to support seamless usage of Y.Js objects in metadata attached to expressions), and AST-based synchronization; integrating the new APIs today. #8825

**Comment by vitvakatu** [comment]
- https://github.com/enso-org/enso/pull/8771
- https://github.com/enso-org/enso/issues/8751
- https://github.com/enso-org/enso/issues/8752

https://github.com/enso-org/enso/assets/6566674/82e1bdb3-3f37-4122-9e73-92d39acd0f13

https://github.com/enso-org/enso/assets/6566674/d8892d00-e6fd-423e-99df-67e0085f6232

**Comment by mwu-tow** [comment]
Last week work:
* Landed https://github.com/enso-org/enso/pull/8762
* Random failure due to httpbin-server failing to start:
  * Landed https://github.com/enso-org/enso/pull/8780
  * Opened https://github.com/enso-org/enso/issues/8796
* Random timeout failure: opened https://github.com/enso-org/enso/issues/8806
* macOS aarch64 build:
  * [WIP] https://github.com/enso-org/enso/pull/8784
  * First packages: https://discord.com/channels/401396655599124480/1197636583835312281

**Comment by JaroslavTulach** [comment]
* Fixed for the most annoying instability bug is ready:  #8787 

![Table_Tests benchmark](https://github.com/enso-org/enso/assets/26887752/5d393f8f-9964-4af1-a5c5-49f2de7186a5)

* Measured **7%** performance improvements of the _`UUID` removal_: #8728
